catalogs/catalog-lakehouse-hudi/src/main/java/org/apache/gravitino/catalog/lakehouse/hudi/backend/hms/HudiHMSTable.java

```diff
@@ -44,15 +49,24 @@ protected HudiHMSTable simpleBuild() {
       table.columns = columns;
       table.indexes = indexes;
       table.partitioning = partitioning;
+      table.sortOrders = sortOrders;
       table.distribution = distribution;
       table.properties = properties;
       table.auditInfo = auditInfo;
       return table;
     }
 
     @Override
-    protected HudiTable buildFromTable(Table backendTable) {
-      // todo: convert HMS table to HudiTable
+    protected HudiHMSTable buildFromTable(Table hmsTable) {
+      name = hmsTable.getTableName();
+      comment = hmsTable.getParameters().get(COMMENT);
+      columns = HiveTableConverter.getColumns(hmsTable, HudiColumn.builder());
+      partitioning = HiveTableConverter.getPartitioning(hmsTable);
+      sortOrders = HiveTableConverter.getSortOrders(hmsTable);
+      distribution = HiveTableConverter.getDistribution(hmsTable);
```

Comment:
Does Hudi store such information in HMS, is it compatible with Hive table? As I know, for Iceberg, we need some Iceberg APIs to get partitioning, sortOrders, because Iceberg will store such information in it's metadata file, not in HMS, I guess Hudi is similar, can you please confirm this?

Reply:
comments added to the code